### ISSUE TITLE
Feature - Add a Helper to manage the HTML with ACL.

### ISSUE THREAD

**Issue by Xety** [cakephp/acl#33]
Hello,

In general, when we use ACL, we also need to do some tests of permissions in the views and display/hide some codes.
For example: 
- A link of a Dashbord (Administration Panel)
- Somes buttons, like `Edit`, `Delete` in a blog or forum
- etc

That's why i was looking if something already existed for Cake2. And i found an awesome helper for Cake1: https://github.com/geromey/acl_utilities

This helper is awesome because it allow us to do ACL tests with 2 differents way.
First :

``` php
<?= $this->Acl->link(
    __('Dashboard'),
    [
        'controller' => 'admin',
        'action' => 'home',
        'prefix' => 'admin'
    ]
) ?>
```

It will look if the current user can access to the url `admin/Admin/home` and if yes, it will display the link, else the link will not be rendered.

Also, sometimes we need to display/hide some HTML code with the link, example with an `<li>` tag :

``` php
<li>
    <?= $this->Acl->link(
        __('Dashboard'),
        [
            'controller' => 'admin',
            'action' => 'home',
            'prefix' => 'admin'
        ]
    ) ?>
</li>
<li>
<!--Another link-->
</li>
etc
```

And this helper provide an awesome method similar to `$this->start()` and `$this->end()` :

``` php
<?php $this->Acl->startBlock() ?>
    <li>
        <?= $this->Acl->link(
            __('Dashboard'),
            [
                'controller' => 'admin',
                'action' => 'home',
                'prefix' => 'admin'
            ]
        ) ?>
    </li>
<?php $this->Acl->endBlock() ?>
```

And if the permissions for the link (Or one of them if there is many link in the same block) between those block are `true`, then all the code will be displayed (Or hidden if the permission for all links are `false`).

I have tried to convert this helper to Cake3, and it work nice but i need to rewrite some codes and fix a bug because if i use 1 time `$this->Acl->startBlock()` in a view it work well, but with 2 or more, none of them work.
But i'm pretty sure that you guys will help me to code it better if the feature is accepted (Example, delete the `_getPatch()` method to use that one from the plugin). :smile: 

Here is my actual helper : https://gist.github.com/Xety/1d590a0fd029a163831a

**Comment by rchavik** [comment]
I like this. Would be interest to see a fleshed out PR on this.

With tests please :)

**Comment by dakota** [comment]
Any chance of creating a PR for this?

**Comment by anhtuank7c** [comment]
@Xety 
@dakota 
How about this issue?

**Comment by shahonseven** [comment]
@Xety I just release AclHelper for CakePHP 3.x
https://github.com/shahonseven/CakePHP-AclLinkHelper/tree/3.x

**Comment by Wouter0100** [comment]
@shahonseven Maybe consider contributing it to this plugin? As this one is fully tested and managed the Cake core team, to ensure quality.

**Comment by dakota** [comment]
@shahonseven I would more more than happy to merge a PR adding a helper like that to the ACL plugin! Provided it had tests, etc ☺

**Comment by shahonseven** [comment]
@dakota since this is my first time contributing to a project, what should I do first?

thanks.

**Comment by Xety** [comment]
@dakota Can you please create a new branch `helper`, so i can push my helper on this branch and work together for the tests.

**Comment by rchavik** [comment]
@Xety I just created the requested `helper` branch

**Comment by Xety** [comment]
Closed as #108 is open now.
